### DISCUSSION TITLE
How to correctly initialise the SDK for iOS and Android?

### DISCUSSION THREAD

**Discussion by walterholohan** [thebergamo/react-native-fbsdk-next#247] [Q&A]
Hi,

I'm just reaching out as I've been struggling to understand if I am initialising the SDK correctly as I've been unsuccessful in testing the logging of app events using the `Text Event` section in `Events Manager`. From reading the documentation I understand there are two ways of initialising the SDK:

1. Natively i.e. adding `[FBSDKApplicationDelegate.sharedInstance initializeSDK]` to `appDelegate.m`
2. Platform-neutral  i.e. Settings.initializeSDK()

I've got a couple of questions about each approach.

### Natively
1. If you initialise the app in appDelegate.m what happens when you need to ask a user for ATT on iOS 14+? Do you need to add `FacebookAdvertiserIDCollectionEnabled = false` to your info.plist and then call `Settings.setAdvertiserTrackingEnabled(true)` after the user has accepted ATT in your app?
2. What about Android? Will this still be auto-init?
3. I noticed in your example app you have the following set in info.plist. Are these necessary?
```
<key>FacebookAutoLogAppEventsEnabled</key>
<string>FALSE</string>
<key>FacebookAdvertiserIDCollectionEnabled</key>
<string>FALSE</string>
```
4. If only using the SDK for logging events do you need to add the below native code to appDelegate.m?
<img width="911" alt="image" src="https://user-images.githubusercontent.com/5293650/166826323-b74697fd-3712-4dc1-8d4f-c9a5d66fc721.png">

### Platform-neutral
1. Do you need to call `Settings.initializeSDK()` for Android?
2. If using `Settings.initializeSDK()` do you need to complete number 4 above i.e. adding the native code to appDelegate.m?
3. Do you need to add the FacebookAdvertiserIDCollectionEnabled to info.plist if you are calling Settings.initializeSDK() after the users accepts ATT?

Just a FYI, I couldn't for the love or money test out logging on Events Manager using the below guide.
<img width="896" alt="image" src="https://user-images.githubusercontent.com/5293650/166826914-19518ea7-f13f-4223-a183-9d9b2d0a0a41.png">

**Comment by mikehardy** [comment]
I must admit, I really don't know personally, App Events aren't a use case for my own apps, however that section has `git blame` attributed to Nick Petersen (@nicknjpconsultingllc / @nickelbob) and perhaps they know and/or have current experience?

**Reply by walterholohan** [reply]
No problem at all @mikehardy, I appreciate the honesty. Hopefully @nicknjpconsultingllc will be able to help with a few of the questions. In the meantime I will continue with doing a deepdive into the native ios facebook sdk repo to see if I can gain more understanding from there.

**Comment by lavpreetsandhu** [comment]
hi @walterholohan you check **[this article](https://dev.to/aneeqakhan/how-to-add-fb-sdk-in-react-native-project-109d)** may be it will help you
